packages/compat/tests/fixup-rules.js

```diff
@@ -133,7 +134,7 @@ describe("@eslint/backcompat", () => {
 			const linter = new Linter();
 			const code = "var foo = () => 123; function bar() { return 123; }";
 			const messages = linter.verify(code, config, {
-				filename: "test.js",
+				filename: path.resolve("test.js"),
```

Comment:
When the `filename` parameter is specified, it will be passed as-is to `configArray.getConfig()`, and `getConfig()` expects a full path:

https://github.com/eslint/rewrite/blob/c94ab2dea6410698a4b6814f921b39774b0e56ab/packages/config-array/src/config-array.js#L970